File: CS/NewObjectFromNavigationExample.Module.Win/Controllers/WinNewObjectFromNavigationController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using NewObjectFromNavigationExample.Module.Controllers;
using DevExpress.ExpressApp.Actions;
using DevExpress.ExpressApp.Win;
using DevExpress.ExpressApp.SystemModule;

namespace NewObjectFromNavigationExample.Module.Win.Controllers {
    public class WinNewObjectFromNavigationController : NewObjectFromNavigationController {
        protected override SingleChoiceAction GetNewObjectAction() {
            MdiShowViewStrategy strategy = Application.ShowViewStrategy as MdiShowViewStrategy;
            if (strategy != null) {
                WinWindow activeInspector = strategy.GetActiveInspector();
                if (activeInspector == null) return null;
                return activeInspector.GetController<NewObjectViewController>().NewObjectAction;
            }
            return base.GetNewObjectAction();
        }
    }
}

File: VB/NewObjectFromNavigationExample.Module.Win/Controllers/WinNewObjectFromNavigationController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports NewObjectFromNavigationExample.Module.Controllers
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.ExpressApp.Win
Imports DevExpress.ExpressApp.SystemModule

Namespace NewObjectFromNavigationExample.Module.Win.Controllers
	Public Class WinNewObjectFromNavigationController
		Inherits NewObjectFromNavigationController
		Protected Overrides Function GetNewObjectAction() As SingleChoiceAction
			Dim strategy As MdiShowViewStrategy = TryCast(Application.ShowViewStrategy, MdiShowViewStrategy)
			If strategy IsNot Nothing Then
				Dim activeInspector As WinWindow = strategy.GetActiveInspector()
				If activeInspector Is Nothing Then
					Return Nothing
				End If
				Return activeInspector.GetController(Of NewObjectViewController)().NewObjectAction
			End If
			Return MyBase.GetNewObjectAction()
		End Function
	End Class
End Namespace
